controllers/clustertopology_controller_sync.go

```diff
@@ -99,3 +108,193 @@ func checkReferencedObjectsAreStrictlyCompatible(current, desired client.Object)
 	}
 	return checkReferencedObjectsAreCompatible(current, desired)
 }
+
+// reconcileMachineDeployments reconciles the desired state of the MachineDeployment objects.
+func (r *ClusterTopologyReconciler) reconcileMachineDeployments(ctx context.Context, current, desired *clusterTopologyState) error {
+	currentMachineDeployments, err := getMachineDeploymentsByID(current.machineDeployments)
+	if err != nil {
+		return err
+	}
+	desiredMachineDeployments, err := getMachineDeploymentsByID(desired.machineDeployments)
+	if err != nil {
+		return err
+	}
+
+	toCreate, toUpdate, toDelete := calculateMachineDeploymentDiff(currentMachineDeployments, desiredMachineDeployments)
```

Comment:
Currently in the WIP for the other reconcile PR we are using reconcile for addressing both create and update.
If we are going to split Create and Update here, this will make a little bit complex to a shared utility for template rotation, which is kind of nice.

Reply:
Yup will merge those and use your util func